compose-test.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ -z "${GITHUB_HEAD_REF##*/}" ]]; then
+  echo "BRANCH_NAME=${GITHUB_REF##*/}" > .env
+else
+  echo "BRANCH_NAME=${GITHUB_HEAD_REF##*/}" > .env
+fi
+docker-compose -f api-tests/docker-compose.yml up -d
+echo "Waiting for API client to discover dependencies and exit"
+sleep 15
+EXIT_CODE=$(docker inspect api-tests_api-client_1 --format='{{.State.ExitCode}}')
```

Comment:
Consider using `docker-compose up --exit-code-from <SERVICE>` instead of `up -d` and `sleep`.
See `docker-compose up --help` for info on the flag

```suggestion
set +e # disable exit on error to be able to catch the exit code from compose
docker-compose -f api-tests/docker-compose.yml up --exit-code-from api-client
set -e # reenable exit on error
EXIT_CODE=$?
```